mlir/lib/Dialect/SCF/IR/SCF.cpp

```diff
@@ -1415,6 +1416,31 @@ InParallelOp ForallOp::getTerminator() {
   return cast<InParallelOp>(getBody()->getTerminator());
 }
 
+FailureOr<SubsetInsertionOpInterface>
+ForallOp::getStoreOpUser(BlockArgument bbArg) {
+  Value::user_range users = bbArg.getUsers();
+  bool foundUser = false;
+  SubsetInsertionOpInterface storeOp = nullptr;
+  for (Operation *userOp : users) {
+    if (auto parallelInsertSliceOp =
+            dyn_cast<tensor::ParallelInsertSliceOp>(userOp);
+        parallelInsertSliceOp && isa<InParallelOp>(userOp->getParentOp())) {
+      // Return failure in case we find more than one user of the block argument
+      // within scf.forall.in_parallel.
+      if (foundUser) {
+        return failure();
+      }
+      storeOp = cast<SubsetInsertionOpInterface>(userOp);
```

Comment:
This should not be necessary.

Reply:
I think you do if you want to go from operation to interface op?

Reply:
What I meant is we don’t need the SubsetInsertionOpInterface in this PR at all. It should be sufficient to check whether the bbarg is used in the terminator region (as a destination) or not. No need to analyze the loop body.